tabular/src/autogluon/tabular/models/xgboost/xgboost_model.py

```diff
@@ -0,0 +1,122 @@
+        if eval_metric is None:
+            eval_metric = xgboost_utils.func_generator(metric=self.stopping_metric, is_higher_better=True, needs_pred_proba=not self.stopping_metric_needs_y_pred, problem_type=self.problem_type)
+        return eval_metric
+
+    def _preprocess(self, X, is_train=False, max_category_levels=None, **kwargs):
+        X = super()._preprocess(X=X, **kwargs)
+
+        if self._ohe_generator is None:
+            self._ohe_generator = xgboost_utils.OheFeatureGenerator(max_levels=max_category_levels)
+
+        if is_train:
+            self._ohe_generator.fit(X)
+
+        X = self._ohe_generator.transform(X)
+
+        return X
+
+    def _fit(self, X_train, y_train, X_val=None, y_val=None, time_limit=None, **kwargs):
+        start_time = time.time()
+        
+        invalid_params = ['num_threads', 'num_gpus']
+        for invalid in invalid_params:
+            if invalid in self.params:
+                self.params.pop(invalid)
+        params = self.params.copy()
+        max_category_levels = params.pop('proc.max_category_levels', 100)
+
+        verbosity = kwargs.get('verbosity', 2)
+        if verbosity <= 2:
+            verbose = False
+        elif verbosity == 3:
+            verbose = True
+            verbose_eval = 50
+        else:
+            verbose = True
+            verbose_eval = 1
+        
+        X_train = self.preprocess(X_train, is_train=True, max_category_levels=max_category_levels)
+        num_rows_train = X_train.shape[0]
+
+        eval_set = []
+        eval_metric = self.get_eval_metric()
+
+        if X_val is None:
+            early_stopping_rounds = 150
+            eval_set.append((X_train, y_train))  # TODO: if the train dataset is large, use sample of train dataset for validation
+        else:
+            modifier = 1 if num_rows_train <= 10000 else 10000 / num_rows_train
+            early_stopping_rounds = max(round(modifier * 150), 10)
+            X_val = self.preprocess(X_val, is_train=False)
+            eval_set.append((X_val, y_val))
+
+        try_import_xgboost()
+        from .callbacks import print_evaluation, early_stop_custom
+        callbacks = []
+        if verbose:
+            callbacks.append(print_evaluation(verbose_eval))
+        callbacks.append(early_stop_custom(early_stopping_rounds, start_time=start_time, time_limit=time_limit, verbose=verbose))
+
+        from xgboost import XGBClassifier, XGBRegressor
+        model_type = XGBClassifier if self.problem_type in PROBLEM_TYPES_CLASSIFICATION else XGBRegressor
+        self.model = model_type(**params)
+        self.model.fit(
+            X=X_train,
+            y=y_train,
+            eval_set=eval_set,
+            eval_metric=eval_metric,
+            verbose=False,
+            callbacks=callbacks
+        )
+
+        bst = self.model.get_booster()
+        self.params_trained['n_estimators'] = bst.best_iteration + 1
+        self.params_trained['best_ntree_limit'] = bst.best_ntree_limit
```

Comment:
This line should be removed since this is not a parameter used for training in refit_full.

Currently during refit_full this message occurs:

```
[12:06:12] WARNING: /Users/travis/build/dmlc/xgboost/src/learner.cc:516: 
Parameters: { best_ntree_limit } might not be used.
  This may not be accurate due to some parameters are only used in language bindings but
  passed down to XGBoost core.  Or some parameters are not used but slip through this
  verification. Please open an issue if you find above cases.
```